src/chat/backend.rs

```diff
@@ -98,19 +112,40 @@ impl CallCodec {
             conn_id: cid,
             destination: destination.clone(),
             sender: sender,
+            request_id_header: request_id_header,
+        }
+    }
+
+    fn add_request_id<S>(&self, e: &mut http::Encoder<S>, cid: &str) {
+        if let Some(ref header) = self.request_id_header {
+            // NOTE: should request_id validation be more strict?
+            let rxid = match self.meta.get("request_id").unwrap() {
+                &Json::String(ref s) if s.len() <= 36 => {
+                    format!("{}-{}", cid, s)
+                }
+                &Json::Number(ref n) if n.is_u64() => {
+                    format!("{}-{}", cid, n.as_u64().unwrap())
+                }
+                _ => {
+                    format!("{}-{}", cid, request_id::new())
+                }
+            };
+            e.add_header(header, rxid).unwrap();
         }
     }
 }
 
 impl InactivityCodec {
     pub fn new(path: &Arc<String>, sid: &SessionId,
         destination: &Arc<Destination>)
+        // request_id_header: Option<String>)
```

Comment:
Hmm, InactivityCodec depends on session pool and not chat handler (settings)
so we either should have another `request-id-header` setting in session-pool, 
or drop it from here.
Maybe we should move `request-id-header` setting from chat & proxy handlers
into http-destination and provide some generic interface to set it for chat/proxy/whatever handlers?